File: src/main/java/com/example/today_i_baekjoon/dto/CommitInfo.java
package com.example.today_i_baekjoon.dto;

import com.example.today_i_baekjoon.domain.Commit;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.ToString;

import java.time.LocalDate;

@ToString
@AllArgsConstructor
@Builder
@Getter
public class CommitInfo {

	private LocalDate commitDate;
	private String commitUrl;
	private String username;

	public static CommitInfo fromEntity(Commit commit) {
		return CommitInfo.builder()
				.commitDate(commit.getCommitDate())
				.commitUrl(commit.getCommitUrl())
				.username(commit.getUser().getUsername())
				.build();
	}
}

File: src/main/kotlin/com/gidskql6671/today_i_baekjoon/dto/CommitInfo.kt
package com.gidskql6671.today_i_baekjoon.dto

import com.gidskql6671.today_i_baekjoon.domain.Commit
import java.time.LocalDate

data class CommitInfo(
    val commitDate: LocalDate,
    val commitUrl: String,
    val username: String,
    val problemTitle: String,
    val problemRank: String
)

fun commitInfoFromEntity(commit: Commit) = CommitInfo(
    commitDate = commit.commitDate,
    commitUrl = commit.commitUrl,
    username = commit.user.username,
    problemTitle = commit.problemTitle,
    problemRank = commit.problemRank
)